Review the documentation regarding entity inheritance (#10429)
doctrine/doctrine2: docs/en/reference/inheritance-mapping.rst

appear in the middle of an otherwise mapped inheritance hierarchy
(through Single Table Inheritance or Class Table Inheritance).

> No database table will be created for a mapped superclasse itself,

Reviewer: ```suggestion
No database table will be created for a mapped superclass itself,
```

descendants of this root entity.

On the root entity class, ``#[InheritanceType]``,
``#[DiscriminatorColumn]`` and ``#[DiscriminatorMap]`` must be specified.

``#[InheritanceType]`` specifies one of the two available inheritance
mapping strategies that are explained in the following sections.

``#[DiscriminatorColumn]`` designates the so-called discriminator column.
This is an extra column in the database that keeps information about which
type from the hierarchy applies for a particular database row.

``#[DiscriminatorMap]`` declares the possible values for the discriminator
column and maps them to class names in the hierarchy. This discriminator map
has to declare all non-abstract entity classes that exist in that particular
inheritance hierarchy. That includes the root as well as any intermediate
entity classes, given they are not abstract.

The names of the classes in the discriminator map do not need to be fully
qualified if the classes are contained in the same namespace as the entity
class on which the discriminator map is applied.

If no discriminator map is provided, then the map is generated
automatically. The automatically generated discriminator map contains the

> lowercase short name of each class as key.

Author: I have kept that, but in fact, I am not sure if it is even true?

Author: It fact it is.

superclass were inherited to the subclass as if they had been
defined on that class directly.

Entity Inheritance
------------------

As soon as one entity class inherits from another entity class, either
directly, with a mapped superclass or other unmapped (also called
"transient") classes in between, these entities form an inheritance
hierarchy. The topmost entity class in this hierarchy is called the
root entity, and the hierarchy includes all entities that are
descendants of this root entity.

On the root entity class, ``#[InheritanceType]``,
``#[DiscriminatorColumn]`` and ``#[DiscriminatorMap]`` must be specified.

``#[InheritanceType]`` specifies one of the two available inheritance
mapping strategies that are explained in the following sections.

``#[DiscriminatorColumn]`` designates the so-called discriminator column.

> This is an extra column in the database that keeps information about which

Reviewer: Let's talk about the table for the root entity class instead of talking about the database? One may think there is only one such column in the whole database.

```suggestion
This is an extra column in the table that keeps information about which
```

type from the hierarchy applies for a particular database row.

``#[DiscriminatorMap]`` declares the possible values for the discriminator
column and maps them to class names in the hierarchy. This discriminator map
has to declare all non-abstract entity classes that exist in that particular
inheritance hierarchy. That includes the root as well as any intermediate
entity classes, given they are not abstract.

The names of the classes in the discriminator map do not need to be fully
qualified if the classes are contained in the same namespace as the entity
class on which the discriminator map is applied.

If no discriminator map is provided, then the map is generated
automatically. The automatically generated discriminator map contains the
lowercase short name of each class as key.

.. note::

>     Automatically generating the descriminator map is very expensive

Reviewer: ```suggestion
    Automatically generating the discriminator map is very expensive
```

    computation-wise. The mapping driver has to provide all classes
    for which mapping configuration exists, and those have to be

>     loaded and checked whether they are part of the current
>     inheritance hierarchy. The resulting map, however, can be kept

Reviewer: ```suggestion
    loaded and checked against  the current inheritance hierarchy
    to see if they are part of it. The resulting map, however, can be kept
```

using either a legacy database schema or a self-written database
schema you have to make sure that all columns that are not in the
root entity but in any of the different sub-entities has to allow
null values. Columns that have ``NOT NULL`` constraints have to be on
the root entity of the single-table inheritance hierarchy.

Class Table Inheritance
-----------------------

`Class Table Inheritance <https://martinfowler.com/eaaCatalog/classTableInheritance.html>`_
is an inheritance mapping strategy where each class in a hierarchy

> is mapped to several tables: Its own table and the tables of all

Reviewer: I don't think that fix is correct: the second part isn't even a valid, standalone sentence (it has no verb). Even if it did, I would keep this lowercase.

Author: okay, will fix
